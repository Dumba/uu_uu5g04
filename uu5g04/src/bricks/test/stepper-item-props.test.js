/**
 * Copyright (C) 2019 Unicorn a.s.
 *
 * This program is free software; you can use it under the terms of the UAF Open License v01 or
 * any later version. The text of the license is available in the file LICENSE or at www.unicorn.com.
 *
 * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even
 * the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See LICENSE for more details.
 *
 * You may contact Unicorn a.s. at address: V Kapslovne 2767/2, Praha 3, Czech Republic or
 * at the email: info@unicorn.com.
 */

import UU5 from "uu5g04";
import "uu5g04-bricks";

const { mount, shallow, wait } = UU5.Test.Tools;

const CONFIG = {
  mixins: ["UU5.Common.BaseMixin", "UU5.Common.ElementaryMixin", "UU5.Common.ColorSchemaMixin"],

  props: {
    size: {
      values: ["s", "m", "l", "xl"],
    },
    borderRadius: {
      values: ["8", "15"],
    },
    bgStyle: {
      values: ["filled", "outline", "transparent", "underline"],
    },
    topVisitedStep: {
      values: [0, 5],
    },
    currentStep: {
      values: [0, 4],
    },
    elevation: {
      values: ["-1", "0", "1", "2", "3", "4", "5", -1, 0, 1, 2, 3, 4, 5],
    },
    index: {
      values: [0, 3],
    },
  },
  requiredProps: {
    parent: shallow(<UU5.Bricks.Stepper id={"idParent"} />).instance(),
  },
};

describe(`UU5.Bricks.Stepper.Item props`, () => {
  UU5.Test.Tools.testProperties(UU5.Bricks.Stepper.Item, CONFIG);
});

describe(`UU5.Bricks.Stepper.Item  props.function `, () => {
  it(`UU5.Bricks.Stepper.Item - onClick()`, () => {
    let mockFn = jest.fn();
    let secondItem;
    let fourItem;
    const wrapper = mount(
      <UU5.Bricks.Stepper id={"uuID"} onClick={mockFn} currentStep={2} topVisitedStep={3}>
        <UU5.Bricks.Stepper.Item>Dokoncený krok</UU5.Bricks.Stepper.Item>
        <UU5.Bricks.Stepper.Item ref_={(comp) => (secondItem = comp)}>Dokoncený krok</UU5.Bricks.Stepper.Item>
        <UU5.Bricks.Stepper.Item>Aktivní krok</UU5.Bricks.Stepper.Item>
        <UU5.Bricks.Stepper.Item ref_={(comp2) => (fourItem = comp2)}>Následujicí krok</UU5.Bricks.Stepper.Item>
        <UU5.Bricks.Stepper.Item>Následujicí krok</UU5.Bricks.Stepper.Item>
      </UU5.Bricks.Stepper>
    );

    wrapper.find(UU5.Bricks.Button).at(1).simulate("click");
    wrapper.find(UU5.Bricks.Button).at(4).simulate("click");

    expect(mockFn).toHaveBeenCalledTimes(1);
    let opt = mockFn.mock.calls[0][0];
    expect(opt).toMatchObject({
      value: 1,
    });
    expect(opt.component === secondItem).toBeTruthy();
    expect(opt.event).toBeTruthy();
    //trying click on disabled step
    expect(opt.component === fourItem).toBeFalsy();
  });
});
