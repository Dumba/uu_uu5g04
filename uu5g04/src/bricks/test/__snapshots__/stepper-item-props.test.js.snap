// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UU5.Bricks.Stepper.Item props bgStyle 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`UU5.Bricks.Stepper.Item props bgStyle 2`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`UU5.Bricks.Stepper.Item props bgStyle 3`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`UU5.Bricks.Stepper.Item props bgStyle 4`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`UU5.Bricks.Stepper.Item props bgStyle 5`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`UU5.Bricks.Stepper.Item props bgStyle 6`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`UU5.Bricks.Stepper.Item props bgStyle 7`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`UU5.Bricks.Stepper.Item props bgStyle 8`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`UU5.Bricks.Stepper.Item props borderRadius 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`UU5.Bricks.Stepper.Item props borderRadius 2`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`UU5.Bricks.Stepper.Item props borderRadius 3`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`UU5.Bricks.Stepper.Item props borderRadius 4`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`UU5.Bricks.Stepper.Item props className 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`UU5.Bricks.Stepper.Item props colorSchema 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`UU5.Bricks.Stepper.Item props colorSchema 2`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`UU5.Bricks.Stepper.Item props colorSchema 3`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`UU5.Bricks.Stepper.Item props colorSchema 4`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`UU5.Bricks.Stepper.Item props colorSchema 5`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`UU5.Bricks.Stepper.Item props colorSchema 6`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`UU5.Bricks.Stepper.Item props colorSchema 7`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`UU5.Bricks.Stepper.Item props colorSchema 8`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`UU5.Bricks.Stepper.Item props colorSchema 9`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`UU5.Bricks.Stepper.Item props colorSchema 10`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`UU5.Bricks.Stepper.Item props colorSchema 11`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`UU5.Bricks.Stepper.Item props colorSchema 12`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`UU5.Bricks.Stepper.Item props colorSchema 13`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`UU5.Bricks.Stepper.Item props colorSchema 14`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`UU5.Bricks.Stepper.Item props colorSchema 15`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`UU5.Bricks.Stepper.Item props colorSchema 16`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`UU5.Bricks.Stepper.Item props colorSchema 17`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`UU5.Bricks.Stepper.Item props colorSchema 18`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`UU5.Bricks.Stepper.Item props colorSchema 19`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`UU5.Bricks.Stepper.Item props colorSchema 20`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`UU5.Bricks.Stepper.Item props colorSchema 21`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`UU5.Bricks.Stepper.Item props colorSchema 22`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`UU5.Bricks.Stepper.Item props colorSchema 23`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`UU5.Bricks.Stepper.Item props colorSchema 24`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`UU5.Bricks.Stepper.Item props colorSchema 25`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`UU5.Bricks.Stepper.Item props colorSchema 26`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`UU5.Bricks.Stepper.Item props colorSchema 27`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`UU5.Bricks.Stepper.Item props colorSchema 28`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`UU5.Bricks.Stepper.Item props colorSchema 29`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`UU5.Bricks.Stepper.Item props colorSchema 30`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`UU5.Bricks.Stepper.Item props colorSchema 31`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`UU5.Bricks.Stepper.Item props colorSchema 32`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`UU5.Bricks.Stepper.Item props colorSchema 33`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`UU5.Bricks.Stepper.Item props colorSchema 34`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`UU5.Bricks.Stepper.Item props colorSchema 35`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`UU5.Bricks.Stepper.Item props colorSchema 36`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`UU5.Bricks.Stepper.Item props colorSchema 37`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`UU5.Bricks.Stepper.Item props colorSchema 38`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`UU5.Bricks.Stepper.Item props controlled 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`UU5.Bricks.Stepper.Item props controlled 2`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`UU5.Bricks.Stepper.Item props controlled 3`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`UU5.Bricks.Stepper.Item props controlled 4`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`UU5.Bricks.Stepper.Item props currentStep 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`UU5.Bricks.Stepper.Item props currentStep 2`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`UU5.Bricks.Stepper.Item props currentStep 3`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`UU5.Bricks.Stepper.Item props currentStep 4`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`UU5.Bricks.Stepper.Item props default props 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`UU5.Bricks.Stepper.Item props disabled 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`UU5.Bricks.Stepper.Item props disabled 2`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`UU5.Bricks.Stepper.Item props disabled 3`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`UU5.Bricks.Stepper.Item props disabled 4`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`UU5.Bricks.Stepper.Item props elevation 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`UU5.Bricks.Stepper.Item props elevation 2`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`UU5.Bricks.Stepper.Item props elevation 3`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`UU5.Bricks.Stepper.Item props elevation 4`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`UU5.Bricks.Stepper.Item props elevation 5`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`UU5.Bricks.Stepper.Item props elevation 6`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`UU5.Bricks.Stepper.Item props elevation 7`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`UU5.Bricks.Stepper.Item props elevation 8`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`UU5.Bricks.Stepper.Item props elevation 9`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`UU5.Bricks.Stepper.Item props elevation 10`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`UU5.Bricks.Stepper.Item props elevation 11`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`UU5.Bricks.Stepper.Item props elevation 12`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`UU5.Bricks.Stepper.Item props elevation 13`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`UU5.Bricks.Stepper.Item props elevation 14`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`UU5.Bricks.Stepper.Item props elevation 15`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`UU5.Bricks.Stepper.Item props elevation 16`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`UU5.Bricks.Stepper.Item props elevation 17`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`UU5.Bricks.Stepper.Item props elevation 18`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`UU5.Bricks.Stepper.Item props elevation 19`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`UU5.Bricks.Stepper.Item props elevation 20`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`UU5.Bricks.Stepper.Item props elevation 21`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`UU5.Bricks.Stepper.Item props elevation 22`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`UU5.Bricks.Stepper.Item props elevation 23`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`UU5.Bricks.Stepper.Item props elevation 24`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`UU5.Bricks.Stepper.Item props elevation 25`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`UU5.Bricks.Stepper.Item props elevation 26`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`UU5.Bricks.Stepper.Item props elevation 27`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`UU5.Bricks.Stepper.Item props elevation 28`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`UU5.Bricks.Stepper.Item props hidden 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`UU5.Bricks.Stepper.Item props hidden 2`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`UU5.Bricks.Stepper.Item props hidden 3`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`UU5.Bricks.Stepper.Item props hidden 4`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`UU5.Bricks.Stepper.Item props id 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`UU5.Bricks.Stepper.Item props index 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`UU5.Bricks.Stepper.Item props index 2`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`UU5.Bricks.Stepper.Item props index 3`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`UU5.Bricks.Stepper.Item props index 4`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`UU5.Bricks.Stepper.Item props mainAttrs 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`UU5.Bricks.Stepper.Item props name 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`UU5.Bricks.Stepper.Item props noIndex 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`UU5.Bricks.Stepper.Item props noIndex 2`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`UU5.Bricks.Stepper.Item props noIndex 3`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`UU5.Bricks.Stepper.Item props noIndex 4`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`UU5.Bricks.Stepper.Item props selected 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`UU5.Bricks.Stepper.Item props selected 2`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`UU5.Bricks.Stepper.Item props selected 3`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`UU5.Bricks.Stepper.Item props selected 4`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`UU5.Bricks.Stepper.Item props size 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`UU5.Bricks.Stepper.Item props size 2`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`UU5.Bricks.Stepper.Item props size 3`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`UU5.Bricks.Stepper.Item props size 4`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`UU5.Bricks.Stepper.Item props size 5`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`UU5.Bricks.Stepper.Item props size 6`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`UU5.Bricks.Stepper.Item props size 7`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`UU5.Bricks.Stepper.Item props size 8`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`UU5.Bricks.Stepper.Item props style 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`UU5.Bricks.Stepper.Item props style 2`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`UU5.Bricks.Stepper.Item props style 3`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`UU5.Bricks.Stepper.Item props style 4`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`UU5.Bricks.Stepper.Item props tooltip 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`UU5.Bricks.Stepper.Item props topVisitedStep 1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`UU5.Bricks.Stepper.Item props topVisitedStep 2`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`UU5.Bricks.Stepper.Item props topVisitedStep 3`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;

exports[`UU5.Bricks.Stepper.Item props topVisitedStep 4`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
